test/gtest/uct/uct_test.cc

```diff
@@ -142,33 +142,56 @@ std::vector<uct_test_base::md_resource> uct_test_base::enum_md_resources() {
 
 uct_test::uct_test() {
     ucs_status_t status;
-    uct_md_attr_t pd_attr;
-    uct_md_h pd;
+    uct_md_attr_t md_attr;
+    uct_md_h md;
 
     status = uct_md_config_read(GetParam()->component, NULL, NULL, &m_md_config);
     ASSERT_UCS_OK(status);
 
     status = uct_md_open(GetParam()->component, GetParam()->md_name.c_str(),
-                         m_md_config, &pd);
+                         m_md_config, &md);
     ASSERT_UCS_OK(status);
 
-    status = uct_md_query(pd, &pd_attr);
+    status = uct_md_query(md, &md_attr);
     ASSERT_UCS_OK(status);
 
-    if (pd_attr.cap.flags & UCT_MD_FLAG_SOCKADDR) {
-        status = uct_md_iface_config_read(pd, NULL, NULL, NULL, &m_iface_config);
+    if (md_attr.cap.flags & UCT_MD_FLAG_SOCKADDR) {
+        status = uct_md_iface_config_read(md, NULL, NULL, NULL, &m_iface_config);
+    } else if (!strcmp(GetParam()->tl_name.c_str(), "sockaddr")) {
+        m_iface_config = NULL;
+        return;
     } else {
-        status = uct_md_iface_config_read(pd, GetParam()->tl_name.c_str(), NULL,
+        status = uct_md_iface_config_read(md, GetParam()->tl_name.c_str(), NULL,
                                           NULL, &m_iface_config);
     }
 
     ASSERT_UCS_OK(status);
-    uct_md_close(pd);
+    uct_md_close(md);
 }
 
 uct_test::~uct_test() {
-    uct_config_release(m_iface_config);
+    ucs_status_t status;
+    uct_md_attr_t md_attr;
+    uct_md_config_t *md_config;
+    uct_md_h md;
+
+    status = uct_md_config_read(GetParam()->component, NULL, NULL, &md_config);
+    EXPECT_TRUE(status == UCS_OK);
+
+    status = uct_md_open(GetParam()->component, GetParam()->md_name.c_str(),
+                         md_config, &md);
+    EXPECT_TRUE(status == UCS_OK);
+    uct_config_release(md_config);
+
+    status = uct_md_query(md, &md_attr);
+
+    if (((md_attr.cap.flags & UCT_MD_FLAG_SOCKADDR)) ||
+        (strcmp(GetParam()->tl_name.c_str(), "sockaddr"))) {
+        uct_config_release(m_iface_config);
```

Comment:
all this needed just to check if need to release `m_iface_config`? can we check it against NULL?